opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp

```diff
@@ -300,7 +315,7 @@ copyToWellState(const MultisegmentWellGeneric<Scalar>& mswell,
         const auto& [Rs, Rv] = well_.rateConverter().inferDissolvedVaporisedRatio
             (rsMax, rvMax, segment_rates.begin() + seg * well_.numPhases());
 
-        if (! FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
+        if (! (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx && FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx))) ) {
```

Comment:
personal preference: use || for compounds like this. it reads easier, ie
```c++
if (!active(gas) || !active(oil))
```

Reply:
yeah, I do think the current way easier for me to think. I know it is personal preference will incorporate your suggestion.